Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -172,10 +170,28 @@ public final class CacheAdvance<T: Codable> {
         try header.checkFile()
 
         var messages = [T]()
-        while let encodedMessage = try reader.nextEncodedMessage() {
-            messages.append(try decoder.decode(T.self, from: encodedMessage))
-        }
+        // There is only one range: | `offsetInFileOfOldestMessage` -> `offsetInFileAtEndOfNewestMessage`|
+        if reader.offsetInFileOfOldestMessage < reader.offsetInFileAtEndOfNewestMessage {
```

Comment:
This is nitty, but, let's put the comment within the `if` statement:
```suggestion
        if reader.offsetInFileOfOldestMessage < reader.offsetInFileAtEndOfNewestMessage {
            // There is only one range: | `offsetInFileOfOldestMessage` -> `offsetInFileAtEndOfNewestMessage` |
```